tests/compile-fail/while_loop.rs

```diff
@@ -53,6 +53,33 @@ fn main() {
     while let Some(x) = y { // no error, obviously
         println!("{}", x);
     }
+
+    let mut iter = 1..20;
+    while let Option::Some(x) = iter.next() { //~ERROR this loop could be written as a `for` loop
+        println!("{}", x);
+    }
+
+    let mut iter = 1..20;
+    while let Some(x) = iter.next() { //~ERROR this loop could be written as a `for` loop
+        println!("{}", x);
+    }
```

Comment:
I guess it's a contrived example, but what if the loop breaks and the `iter` is reused? I think `for` loops usually consume the iterator, am I right?

We should at least have a test case.